lib/config/rule-validator.js

```diff
@@ -0,0 +1,155 @@
+/**
+ * @fileoverview Rule Validator
+ * @author Nicholas C. Zakas
+ */
+
+"use strict";
+
+//-----------------------------------------------------------------------------
+// Requirements
+//-----------------------------------------------------------------------------
+
+const ajv = require("../shared/ajv")();
+
+//-----------------------------------------------------------------------------
+// Helpers
+//-----------------------------------------------------------------------------
+
+/**
+ * Finds a rule with the given ID in the given config.
+ * @param {string} ruleId The ID of the rule to find.
+ * @param {Object} config The config to search in.
+ * @returns {{create: Function, schema: (Array|null)}} THe rule object.
+ */
+function findRuleDefinition(ruleId, config) {
+    const ruleIdParts = ruleId.split("/");
+    let pluginName, ruleName;
+
+    // built-in rule
+    if (ruleIdParts.length === 1) {
+        pluginName = "@";
+        ruleName = ruleIdParts[0];
+    } else {
+        ruleName = ruleIdParts.pop();
+        pluginName = ruleIdParts.join("/");
+    }
+
+    if (!config.plugins || !config.plugins[pluginName]) {
+        throw new TypeError(`Key "rules": Key "${ruleId}": Could not find plugin "${pluginName}".`);
+    }
+
+    if (!config.plugins[pluginName].rules || !config.plugins[pluginName].rules[ruleName]) {
+        throw new TypeError(`Key "rules": Key "${ruleId}": Could not find "${ruleName}" in plugin "${pluginName}".`);
+    }
```

Comment:
Should we have a different message for built-in rules, this might be confusing:

`TypeError: Key "rules": Key "no-undeff": Could not find "no-undeff" in plugin "@".`

Reply:
This is something we can address later. I don't want to tie up this PR one small UX discussions like this.